src/spec_tools/json_schema.cljc

```diff
@@ -202,7 +202,10 @@
 (defmethod accept-spec 'clojure.spec/and [dispatch spec children]
   (simplify-all-of {:allOf children}))
 
-; merge
+(defmethod accept-spec 'clojure.spec/merge [dispatch spec children]
+  {:type "object"
+   :properties (apply merge (map :properties children))
+   :required (apply concat (map :required children))})
```

Comment:
When several children define the same key the mapping from the latter will be in the result, same as in clojure.core/merge. Will fix required to be a vec, and add test for case with same keys.